pilot/pkg/bootstrap/server.go

```diff
@@ -504,43 +507,93 @@ func (s *Server) initGrpcServer(options *istiokeepalive.Options) {
 	s.EnvoyXdsServer.Register(s.grpcServer)
 }
 
-// initialize secureGRPCServer
-func (s *Server) initSecureGrpcServer(options *istiokeepalive.Options) error {
+func (s *Server) generateKeyAndCert(saName string, saNamespace string) ([]byte, []byte, error) {
```

Comment:
It's probably ok for this PR, given the goal is to allow smooth upgrade. 

Technically we should use the so-called cache - that we use to generate certs for workloads.
That means if an external CA is used - we'll also use it for the Istiod cert.

But it's probably too much for a one-time upgrade ( since 1.6 upgrade will not need this code )